client/src/components/layout/Header.vue

```diff
@@ -20,7 +20,7 @@
 
       <!-- Right aligned nav items -->
       <b-navbar-nav class="ml-auto">
-        <b-nav-item href="/cart">Cart</b-nav-item>
+        <b-nav-item to="cart">Cart</b-nav-item>
```

Comment:
I can't seem to find where I read it, but there's documentation somewhere for the Bootstrap-Vue library that says `href` generates a normal link and `to` generates a router link.

The normal link was refreshing the page and clearing your state as you moved to the cart page. You remained logged in because the token is stored in the cookie, but the cart isn't persisted anywhere. Router links don't reach out to the browser's level, they only work with the Vue router to adjust the elements on the page as needed. So with router links there isn't any full refresh.

None of that matters here when the modal gets set up, but it'll be a nicer experience to use router links to navigate around.